### PR TITLE
Describe Performance Tracing in the Performance Page

### PR DESCRIPTION
**Pull request by DanielRosenwasser**

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
CC @salemhilal, I think Performance Tracing might be a reasonable way to diagnose where a lot of time is being spent in type-checking.

File: Performance-Tracing.md [hunk 1]
```diff
@@ -13,8 +13,8 @@ If you do share them, please consider zipping them — they compress very well.
 At present, the output is not very user friendly, but here are some steps you can take to identify the parts of your code that are slowing the compiler down.
 The goal of this process is to be able to extract a reduced repro for which you can file an [issue](https://github.com/microsoft/TypeScript/issues).
 
-1.  This document is specific to TS 4.1 RC and TS 4.1, so you'll need to install one of those.
-    The easiest way is to `npm install typescript@rc` (or `typescript@4.1`, once it is released).
+1.  This document is specific to TypeScript 4.1 and higher, so you'll need that first.
```
Comment (amcasey) [inline Performance-Tracing.md:16]: No, it specifically isn't "and higher".

File: Performance.md [hunk 2]
```diff
@@ -481,6 +481,31 @@ Sometimes TypeScript's type-checking can be impacted by computationally intensiv
 This is rare, but can happen.
 Upgrading to a newer version of TypeScript (which can be more efficient) or to a newer version of an `@types` package (which may have reverted a regression) can often solve the issue.
 
+## Performance Tracing
+
+You might find that these diagnostic methods are too opaque to get a sense of why TypeScript feels slow.
```
Comment (amcasey) [inline Performance.md:486]: I don't know what "opaque" means in this context, but it feels like "vague" or "imprecise" might be more appropriate?

File: Performance.md [hunk 3]
```diff
@@ -481,6 +481,31 @@ Sometimes TypeScript's type-checking can be impacted by computationally intensiv
...
+TypeScript provides a `--generateTrace` option that can give you a sense of the work the compiler is spending time on.
+`--generateTrace` will create an output file that can be analyzed within Edge or Chrome.
+
+Make sure TypeScript can compile your project without errors.
```
Comment (amcasey) [inline Performance.md:490]: It actually doesn't matter if there are errors (though you'll be measuring something different).

File: Performance.md [hunk 4]
```diff
@@ -534,6 +559,12 @@ A compiler perf issue should include
 * Ideally, a project that demonstrates the issues being encountered.
 * Output logs from profiling the compiler (`isolate-*-*-*.log` and `*.cpuprofile` files)
 
+### Providing Performance Traces
+
+[Performance traces (described above)](#performance-tracing) are meant to help teams figure out build performance issues in their own codebases;
+however, they can also be useful for the TypeScript team in diagnosing and fixing issues.
+[See the above section on performance traces](#performance-tracing) and continue reading more on [our dedicated performance tracing page](https://github.com/microsoft/TypeScript/wiki/Performance-Tracing).
```
Comment (amcasey) [inline Performance.md:566]: Those are some big link spans. 😛

**Review by amcasey — CHANGES_REQUESTED** [review]

File: Performance-Tracing.md (hunk 1 above)
Comment (DanielRosenwasser) [reply Performance-Tracing.md:16]:
```suggestion
1.  This document is specific to TypeScript 4.1, so you'll need that first.
```

File: Performance.md (hunk 2 above)
Comment (DanielRosenwasser) [reply Performance.md:486]:
```suggestion
In some cases, the approaches above might not give you enough insight to understand why TypeScript feels slow.
```

File: Performance.md [hunk 5]
```diff
@@ -481,6 +481,31 @@ Sometimes TypeScript's type-checking can be impacted by computationally intensiv
 This is rare, but can happen.
 Upgrading to a newer version of TypeScript (which can be more efficient) or to a newer version of an `@types` package (which may have reverted a regression) can often solve the issue.
 
+## Performance Tracing
+
+You might find that these diagnostic methods are too opaque to get a sense of why TypeScript feels slow.
+TypeScript provides a `--generateTrace` option that can give you a sense of the work the compiler is spending time on.
```
Comment (DanielRosenwasser) [inline Performance.md:487]:
```suggestion
TypeScript 4.1 and higher provides a `--generateTrace` option that can give you a sense of the work the compiler is spending time on.
```

File: Performance.md (hunk 3 above)
Comment (DanielRosenwasser) [reply Performance.md:490]:
```suggestion
Ideally, TypeScript will be able to compile your project without any errors, though it's not a strict requirement for tracing.
```

File: Performance.md [hunk 6]
```diff
@@ -481,6 +481,31 @@ Sometimes TypeScript's type-checking can be impacted by computationally intensiv
...
+`--generateTrace` will create an output file that can be analyzed within Edge or Chrome.
+
+Make sure TypeScript can compile your project without errors.
+Once you can get a clean compile/type-check, you can run with the `--generateTrace` flag to specify an output directory for tracing.
```
Comment (DanielRosenwasser) [inline Performance.md:491]:
```suggestion
Once you're ready to get a trace, you can run TypeScript with the `--generateTrace` flag.
```

**Pull request merged**